lte/gateway/c/session_manager/SessionState.cpp

```diff
@@ -2784,4 +2784,212 @@ bool operator==(const Teids& lhs, const Teids& rhs) {
   return lhs.enb_teid() == rhs.enb_teid() && lhs.agw_teid() == rhs.agw_teid();
 }
 
+void SessionState::process_5g_rules_to_install(
+    const std::vector<StaticRuleInstall>& static_rule_installs,
+    const std::vector<DynamicRuleInstall>& dynamic_rule_installs,
+    RulesToProcess* pending_activation, RulesToProcess* pending_deactivation,
+    SessionStateUpdateCriteria* session_uc) {
+  process_static_5g_rule_installs(
+      static_rule_installs, pending_activation, pending_deactivation,
+      session_uc);
+  process_dynamic_5g_rule_installs(
+      dynamic_rule_installs, pending_activation, pending_deactivation,
+      session_uc);
+}
+
+void SessionState::process_static_5g_rule_installs(
```

Comment:
you can reuse `process_static_rule_installs`

Check where `lifetime` comes. I think if the object doesn't exist in the rule we create a default one.


My advice is to use the IDE to run the tests. If you use the debugger you will see step by step the creation of the rules